blaze/expr/collections.py

```diff
@@ -100,10 +100,32 @@ def sort(child, key=None, ascending=True):
     ascending : bool, optional
         Determines order of the sort
     """
+    if ascending not in (True, False):
+        msg = "ascending must be True or False, given {}"
+        raise ValueError(msg.format(ascending))
     if not isrecord(child.dshape.measure):
-        key = None
+        if key is None or isinstance(key, _strtypes):
+            # Handle this case separately.
+            return Sort(child, None, ascending)
```

Comment:
I thought of doing that and I see your case.  What made me go this route was the many spots in the testsuite where we sort like:

```
t.name.sort('name')
```

And if that now gives an exception like the above, I can imagine it would be surprising to users.  I'm undecided here.  Ideally we'd be able to extract the column name from `child` and compare to `key` and validate, but that seems like it would require a lot of work with expression dshapes, and would probably break other code that depends on `t.name` being non-record dshape.  Thoughts?